pkg/pgmodel/migrations/sql/idempotent/base.sql

```diff
@@ -144,6 +141,26 @@ COMMENT ON PROCEDURE SCHEMA_CATALOG.finalize_metric_creation()
 IS 'Finalizes metric creation. This procedure should be run by the connector automatically';
 GRANT EXECUTE ON PROCEDURE SCHEMA_CATALOG.finalize_metric_creation() TO prom_writer;
 
+-- in early versions of timescale multinode we could not enable compression in
+-- functions, so this function allows us to detect non-finalized tables and
+-- enable compression in a seperate statement
+-- we can't use the normal timescaledb info views becaue they also don't work
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.needs_compression_enable()
+RETURNS TEXT[] AS $func$
+DECLARE
+    tables TEXT[];
+BEGIN
+    IF SCHEMA_CATALOG.is_timescaledb_installed() AND SCHEMA_CATALOG.is_multinode() THEN
+        SELECT array_agg(format('%I', table_name))
+        FROM SCHEMA_CATALOG.metric
+        WHERE NOT creation_completed
+        INTO tables;
+    END IF;
+
+    RETURN tables;
```

Comment:
is this guaranteed empty at initialization?

Reply:
I believe it should be NULL at init